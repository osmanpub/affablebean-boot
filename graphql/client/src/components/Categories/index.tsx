import { useQuery } from "@apollo/react-hooks";
import React from "react";
import { getId } from "../../helpers/utils";
import { CategoryState } from "../../interfaces/categories";
import { GET_CATEGORIES } from "../../queries";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome,
} from "./Categories.styles";

function Categories() {
  const { data } = useQuery(GET_CATEGORIES);

  if (!data) {
    return null;
  }

  const { categories } = data;

  if (!categories || !categories.length) {
    return null;
  }

  const categoriesList = categories.map((category: CategoryState) => (
    <Category key={getId(category)} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting data-cy="home-welcome" data-testid="home-welcome">
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categoriesList}</CategoriesRight>
    </div>
  );
}

export default Categories;
