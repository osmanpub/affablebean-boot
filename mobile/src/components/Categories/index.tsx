import React from 'react';
import {FlatList, View} from 'react-native';
import {connect} from 'react-redux';
import {getId} from '../../helpers/utils';
import {
  Categories as CategoriesState,
  CategoryState,
} from '../../interfaces/categories';
import {RootState} from '../../redux';
import Category from '../Category';

type Props = {
  categories: CategoriesState;
  setCategoryProduct: Function;
};

function Categories(props: Props) {
  const {categories, setCategoryProduct} = props;
  const {items} = categories;

  if (items.length === 0) {
    return null;
  }

  return (
    <View
      style={{
        alignItems: 'center',
      }}>
      <FlatList
        data={items}
        keyExtractor={(item: CategoryState) => getId(item).toString()}
        renderItem={({item}) => (
          <Category category={item} setCategoryProduct={setCategoryProduct} />
        )}
      />
    </View>
  );
}

const mapStateToProps = (state: RootState) => ({
  categories: state.categories,
});

export default connect(mapStateToProps)(Categories);
