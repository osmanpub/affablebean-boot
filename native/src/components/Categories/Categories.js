import React from "react";
import PropTypes from "prop-types";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";

export function Categories(props) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  const categories = props.categories.map(category => (
    <Category key={category._links.self.href} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categories}</CategoriesRight>
    </div>
  );
}

Categories.propTypes = {
  categories: PropTypes.array.isRequired
};
