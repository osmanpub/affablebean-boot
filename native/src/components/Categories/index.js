import React from "react";
import { FlatList, View } from "react-native";
import PropTypes from "prop-types";
import Category from "../Category";

export default function Categories(props) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  return (
    <View
      style={{
        alignItems: "center",
        flex: 1
      }}
    >
      <FlatList
        data={props.categories}
        keyExtractor={(item, index) => item._links.self.href}
        renderItem={({ item }) => <Category category={item} />}
      />
    </View>
  );
}

Categories.propTypes = {
  categories: PropTypes.array.isRequired
};
