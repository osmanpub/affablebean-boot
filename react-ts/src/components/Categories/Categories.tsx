import React from "react";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";
import { Category as ICategory } from "../../interfaces/categories";
import { CategoriesProps } from "../../interfaces/props";

export function Categories(props: CategoriesProps) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  const categories = props.categories.map((category: ICategory) => (
    <Category {...category} key={category.id} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>
      <CategoriesRight>{categories}</CategoriesRight>
    </div>
  );
}
