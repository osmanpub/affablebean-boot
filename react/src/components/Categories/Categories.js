import React from "react";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";

export function Categories(props) {
  if (props == null || props.categories.length === 0) {
    return null;
  }

  const categories = props.categories.map(category => (
    <Category key={category._links.self.href} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categories}</CategoriesRight>
    </div>
  );
}
