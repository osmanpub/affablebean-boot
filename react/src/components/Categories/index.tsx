import React from "react";
import { CategoryState } from "../../redux/categories";
import Category from "../Category";
import {
  CategoriesGreeting,
  CategoriesLeft,
  CategoriesRight,
  CategoriesWelcome
} from "./Categories.styles";

type Props = {
  categories: Array<CategoryState>;
};

export default function Categories(props: Props) {
  const { categories } = props;

  if (!categories || categories.length === 0) {
    return null;
  }

  const categoriesList = categories.map((category: CategoryState) => (
    <Category key={category._links.self.href} category={category} />
  ));

  return (
    <div>
      <CategoriesLeft>
        <CategoriesWelcome>
          <CategoriesGreeting>
            Welcome to the online home of the Affable Bean Green Grocer.
          </CategoriesGreeting>
          <p>
            Our unique home delivery service brings you fresh organic produce,
            dairy, meats, breads and other delicious and healthy items direct to
            your doorstep.
          </p>
        </CategoriesWelcome>
      </CategoriesLeft>

      <CategoriesRight>{categoriesList}</CategoriesRight>
    </div>
  );
}
