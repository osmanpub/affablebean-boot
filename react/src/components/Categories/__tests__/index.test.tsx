import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Categories from "..";
import { BrowserRouter as Router } from "react-router-dom";

const mockStore = configureStore([]);

const initialState = {
  categories: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        id: 1,
        name: "dairy"
      },
      {
        id: 2,
        name: "meats"
      },
      {
        id: 3,
        name: "bakery"
      },
      {
        id: 4,
        name: "fruit & veg"
      },
      {
        id: 5,
        name: "cereals"
      },
      {
        id: 6,
        name: "drinks"
      }
    ]
  }
};

const store = mockStore(initialState);

describe("<Categories />", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Categories />
          </Provider>
        </Router>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
