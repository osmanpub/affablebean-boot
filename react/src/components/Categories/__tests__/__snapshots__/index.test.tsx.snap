// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders correctly 1`] = `
<div>
  <div
    className="sc-fzXfNg crhOSq"
  >
    <div
      className="sc-fzXfNi kwZMKk"
    >
      <p
        className="sc-fzXfNf fsXfJA"
        data-cy="home-welcome"
      >
        Welcome to the online home of the Affable Bean Green Grocer.
      </p>
      <p>
        Our unique home delivery service brings you fresh organic produce, dairy, meats, breads and other delicious and healthy items direct to your doorstep.
      </p>
    </div>
  </div>
  <div
    className="sc-fzXfNh gZLYOI"
  >
    <span
      className="categoryBox"
    >
      <a
        href="/category/1"
        onClick={[Function]}
      >
        <span
          className="categoryLabel"
        />
        <span
          className="categoryLabelText"
        >
          dairy
        </span>
        <img
          alt="dairy"
          className="categoryImage"
          data-cy="category-dairy"
          src="/static/img/categories/dairy.jpg"
        />
      </a>
    </span>
    <span
      className="categoryBox"
    >
      <a
        href="/category/2"
        onClick={[Function]}
      >
        <span
          className="categoryLabel"
        />
        <span
          className="categoryLabelText"
        >
          meats
        </span>
        <img
          alt="meats"
          className="categoryImage"
          data-cy="category-meats"
          src="/static/img/categories/meats.jpg"
        />
      </a>
    </span>
    <span
      className="categoryBox"
    >
      <a
        href="/category/3"
        onClick={[Function]}
      >
        <span
          className="categoryLabel"
        />
        <span
          className="categoryLabelText"
        >
          bakery
        </span>
        <img
          alt="bakery"
          className="categoryImage"
          data-cy="category-bakery"
          src="/static/img/categories/bakery.jpg"
        />
      </a>
    </span>
    <span
      className="categoryBox"
    >
      <a
        href="/category/4"
        onClick={[Function]}
      >
        <span
          className="categoryLabel"
        />
        <span
          className="categoryLabelText"
        >
          fruit & veg
        </span>
        <img
          alt="fruit & veg"
          className="categoryImage"
          data-cy="category-fruit & veg"
          src="/static/img/categories/fruit & veg.jpg"
        />
      </a>
    </span>
    <span
      className="categoryBox"
    >
      <a
        href="/category/5"
        onClick={[Function]}
      >
        <span
          className="categoryLabel"
        />
        <span
          className="categoryLabelText"
        >
          cereals
        </span>
        <img
          alt="cereals"
          className="categoryImage"
          data-cy="category-cereals"
          src="/static/img/categories/cereals.jpg"
        />
      </a>
    </span>
    <span
      className="categoryBox"
    >
      <a
        href="/category/6"
        onClick={[Function]}
      >
        <span
          className="categoryLabel"
        />
        <span
          className="categoryLabelText"
        >
          drinks
        </span>
        <img
          alt="drinks"
          className="categoryImage"
          data-cy="category-drinks"
          src="/static/img/categories/drinks.jpg"
        />
      </a>
    </span>
  </div>
</div>
`;
